### PR TITLE
fix: properly encode the LNURLp payerdata JSON

### PR DESCRIPTION
**Pull request by bumi**
Maybe this worked because the old axios version did that automatically (which was wrong)

### Describe the changes you have made in this PR

I think the updated axios version #1702 behaves differently. 
We had passed in the query params here wrong and did not encode the JSON manually. 
Thus the payerdata was sent wrong to the server

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

send a payment to a LN address that supports payerdata.

### CODE REVIEW

File: src/app/screens/LNURLPay/index.tsx [hunk 1]
```diff
@@ -106,8 +106,8 @@ function LNURLPay() {
       // Get the invoice
       const params = {
         amount: parseInt(valueSat) * 1000, // user specified sum in MilliSatoshi
-        comment, // https://github.com/fiatjaf/lnurl-rfc/blob/luds/12.md
-        payerdata, // https://github.com/fiatjaf/lnurl-rfc/blob/luds/18.md
+        comment: comment ? comment : null, // https://github.com/fiatjaf/lnurl-rfc/blob/luds/12.md
+        payerdata: payerdata ? JSON.stringify(payerdata) : null, // https://github.com/fiatjaf/lnurl-rfc/blob/luds/18.md
```
Comment (escapedcat) [inline src/app/screens/LNURLPay/index.tsx:110]: You might be able to write it like this:
```js
      const params = {
        amount: parseInt(valueSat) * 1000, // user specified sum in MilliSatoshi
        ...(comment && { comment }), // https://github.com/fiatjaf/lnurl-rfc/blob/luds/12.md
        ...(payerdata && { payerdata: JSON.stringify(payerdata) }), // https://github.com/fiatjaf/lnurl-rfc/blob/luds/18.md
      };
```
Can `params` be typed?

Comment (bumi) [reply src/app/screens/LNURLPay/index.tsx:110]: how, where, what? 
those params are only used in that call. why/how would I type it?

Comment (escapedcat) [reply src/app/screens/LNURLPay/index.tsx:110]: > why/how would I type it?

To indicate what's required and what's optional for example. To make it easier for other devs. Just a thought. No need block this.

Comment (bumi) [reply src/app/screens/LNURLPay/index.tsx:110]: yes, sure. but how/why would that work here? 

I am passing an amount, comment and payerdata... do you have code?

Comment (escapedcat) [reply src/app/screens/LNURLPay/index.tsx:110]: Simple way might be just this?:

```ts
      const params: {
        amount: number;
        comment?: string;
        payerdata?: string;
      } = {
        amount: parseInt(valueSat) * 1000, // user specified sum in MilliSatoshi
        ...(comment && { comment }), // https://github.com/fiatjaf/lnurl-rfc/blob/luds/12.md
        ...(payerdata && { payerdata: JSON.stringify(payerdata) }), // https://github.com/fiatjaf/lnurl-rfc/blob/luds/18.md
      };
```

Comment (bumi) [reply src/app/screens/LNURLPay/index.tsx:110]: what does this do?
for me that's writing the same thing twice.

Comment (escapedcat) [reply src/app/screens/LNURLPay/index.tsx]:
```ts
        amount: number;
        comment?: string;
        payerdata?: string;
```

You mean this? It's an "inline" type indicating what `params` accepts, how this needs to look like, what's required and what's optional.

**Pull request merged**